components/merkledb/src/group.rs

```diff
@@ -106,11 +107,31 @@ where
     }
 }
 
+impl<T, K, I> Group<T, K, I>
+where
+    T: RawAccess,
+    K: BinaryKey + Clone,
+    I: FromAccess<T>,
+{
+    /// Return all the keys from this group.
+    ///
+    /// Note: use this method carefully, because storing all keys in memory may
+    /// consume a large amount of RAM.
+    pub fn keys<V: BinaryValue>(&self) -> Vec<K> {
```

Comment:
1. What does the `V` type param represent?

2. The method should return `K::Owned`. I guess this is the reason you didn't use `Group<_, str, _>` in the tests.

3. Nit / could be solved separately: It *could* make sense to provide API like

```rust
fn keys_unchecked(&self) -> impl Iterator<Item = K>
```

provided that it's clearly documented that creating any new index while the iterator is in scope may lead to a panic. Or it could be worth providing the iterator for the readonly accesses only (e.g., for `T: AsReadonly<Readonly = T>`) 🤔